src/device/device_ledger.cpp

```diff
@@ -320,7 +320,9 @@ namespace hw {
     bool device_ledger::reset() {
       reset_buffer();
       int offset = set_command_header_noopt(INS_RESET);
-      memmove(this->buffer_send+offset, MONERO_VERSION, strlen(MONERO_VERSION));
+      const size_t verlen = strlen(MONERO_VERSION);
+      ASSERT_X(offset + verlen < BUFFER_SEND_SIZE, "MONERO_VERSION is too long")
```

Comment:
```suggestion
      ASSERT_X(offset + verlen <= BUFFER_SEND_SIZE, "MONERO_VERSION is too long")
```

Reply:
Still believe we have to use `<=`. Does Coverity also complain on `<=` check?

Reply:
I don't know, but I doubt it since <= appears correct. I'd missed this in the mostly pointless discussion about strlen/sizeof.

Reply:
The strlen/sizeof discussion may have been a trivial example, but it reflects a trend that you've also remarked on.  https://github.com/monero-project/monero/pull/5793#issuecomment-525254218

Using PREPROCESSOR_MACROS() makes it very obvious what is happening at compile time. Relying on smarter versions of compilers to convert runtime function() invocations into compile-time constants just obfuscates intent. It obfuscates whether the programmer is actually aware of what they're doing, and it forces later developers to also waste brain cycles wondering about something before moving on to more pressing matters. This is just code hygiene - constants should be obviously constant.